### ISSUE TITLE
Levels Not Saving for LAN Players

### ISSUE THREAD

**Issue by MokahTGS** [BeetoGuy/LevelUp2#31]
When using Levels in a LAN environment (Hosted Single Player and connected LAN players) the host player can use levels normally and everything saves fine between game sessions.  LAN players however can use the levels but when they next log in the level progress is gone along with their experience levels.

Forge 2582
Levels v2.1.1.14

**Comment by BeetoGuy** [comment]
I'm not sure if this would be an issue on my end. I just run on the assumption that if it works on a server, then it should work on LAN, but maybe Forge capabilities don't account for LAN connections?

**Comment by MokahTGS** [comment]
In my experience LAN and server are often treated differently.  At least as far as Minecrafts code and how our experience has been.

**Comment by BeetoGuy** [comment]
Yeah, if that's the case, then I would say that opening to LAN isn't recommended.

**Comment by MokahTGS** [comment]
Um...not really a solution.

**Comment by BeetoGuy** [comment]
The problem is, it isn't really something that I can solve on my end. Capabilities not working on LAN seems to be a Forge issue. The only thing you would be able to do to make things work properly in the meantime would be to run the server as a separate process.

**Comment by spacewolfcub** [comment]
I've seen this sort of problem at the Better Questing github and others. Player IDs are different while playing in LAN than playing online, so unless mojang changes things in the future, mod makers are kind of stuck.

A possible solution is to run a server instance separately on your machine, then also run a client instance and connect to the server. Other players can then connect to the server through the IP, even if they're sitting near you (it would probably change IP every once in a while so you would have to keep giving them the new IP), so they are authenticated through mojang as online players. It should allow Level Up, Better Questing and other mods to work as intended since their IDs are properly supported between sessions as online players.

It shouldn't be more of a strain on your machine than the usual LAN play, since even in single-player you are actually running a server and a client at the same time.
